lib/loaders/url.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "open-uri"
+
+module Loaders
+  class URL < Base
+    # We only look for headings and paragraphs
+    TEXT_CONTENT_TAGS = %w[h1 h2 h3 h4 h5 h6 p]
```

Comment:
What about `<span>` tags?

Reply:
There may be a ton more fringe cases where people use `<article>` (other HTML5 tags) to add content.

Reply:
But again -- perhaps it's a future near-future concern not an immediate one.

Reply:
I've been thinking about this. And for now I think the above is optimal. 
`spans` are too granular. I tried having them and I was just getting a lot of clutter (random words that were just wrapped into span on the page for whatever reason.
`article` (and similar) are superior to paragraph. So, if the page markup is semantic, then article should contain paragraphs and we don't need to query `articles` we just focus on the content. If the markup is not good and tags are used randomly, then we'll loose some of the info. But I think it is better for now to have less important information than all of the text from the page which is just random bits squashed together.

Reply:
This all gets quite interesting when you really dig in. It hasn't been updated for a while, but https://github.com/cantino/ruby-readability might be the ticket to just work at an even higher level than nokogiri directly and is going to do a better job than just pulling inner html on a select set of tags.

Reply:
@rickychilcott Nice fine, we should evaluate this gem. It looks like it's covering a ton of edge cases that would take a long time to develop on our own.